pkg/controller/endpointslice/reconciler.go

```diff
@@ -287,6 +289,13 @@ func (r *reconciler) reconcileByPortMapping(
 			}
 		}
 
+		// If the slice does not have the labels of the service,
+		// update it with the new labels
+		svcLabelsUpdated := !containsLabels(existingSlice.Labels, service.Labels)
+		if svcLabelsUpdated {
+			setServiceLabels(existingSlice, service)
+		}
```

Comment:
If a Service happens to have one of the reserved labels (`discovery.LabelServiceName`, `discovery.LabelManagedBy`) set, I think we're at risk of `svcLabelsUpdated` always being true even if nothing is actually changing. I'm not sure what the best way to handle these edge cases is, but one option might be to include this logic in `setServiceLabels` so we don't miss this edge case in the future if we ever add another reserved label.

Reply:
> If a Service happens to have one of the reserved labels (discovery.LabelServiceName, discovery.LabelManagedBy)

then, the service and the slices will have exactly the same labels, right? svcLabelsUpdated will be false in that case. or I misunderstanding the scenario?

Reply:
I don't think it's very likely, but someone might configure a Service like this:

```
name: "foo"
labels:
  kubernetes.io/service-name: "bar"
```

The corresponding EndpointSlice(s) would have a service name label that matched the Service's actual name, not the label on the Service:
```
name: "foo-1234"
labels:
  kubernetes.io/service-name: "foo"
```

In that scenario I believe `svcLabelsUpdated` would always return true even if nothing was actually going to be changed.

Reply:
omg, you are absolutely right,

Reply:
I was thinking more about this and I think that we should keep it this way, mainly because otherwise the "wrong behavior" will be completely unnoticed, at least this way there is an opportunity to trace it back.
No strong opinion though

Reply:
I'm not quite sure what you mean here. Copy the reserved labels from Service to EndpointSlices? I think that would have pretty unfortunate side affects, including this controller thinking it doesn't manage the EndpointSlice it just copied and instead copying a new one on each update. Kube-Proxy also relies on the value of the Service name label here so if the wrong one were set we would potentially end up with EndpointSlices being tied to a different Service (https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/endpointslicecache.go#L341).

Reply:
yep, I really think we should warn that the service has that label, but generating extra updates is not the right way.

Reply:
updated the pr with a warning and avoiding to update in case the lables of the slice are not updated